File: support/src/main/java/com/bhm/sdk/support/IRequest.java
package com.bhm.sdk.support

import com.bhm.sdk.support.DownloadConfig

/**
 * @author Buhuiming
 * @date :2022/10/19 15:21
 */
interface IRequest {
    fun newRequest(config: DownloadConfig?)
    fun startDownload(url: String?): Boolean
    fun startAllDownloads(): Boolean
    fun pauseDownload(url: String?): Boolean
    fun pauseAllDownloads(): Boolean
    fun removeDownload(url: String?): Boolean
    fun removeAllDownloads(): Boolean
}

File: support/src/main/java/com/bhm/sdk/support/IRequest.kt
package com.bhm.sdk.support

/**
 * @author Buhuiming
 * @date :2022/10/19 15:21
 */
interface IRequest {
    fun newRequest(config: DownloadConfig)
    fun startDownload(url: String, callBack: (DownloadCallBack.() -> Unit)?): Boolean?
    fun reStartDownload(url: String, callBack: (DownloadCallBack.() -> Unit)?): Boolean?
    fun pauseDownload(url: String, callBack: (DownloadCallBack.() -> Unit)?): Boolean?
    fun deleteDownload(url: String, callBack: (DownloadCallBack.() -> Unit)?): Boolean?
}